File: src/020/p020.erl
-module(p020).
-export([solve/0]).

factorial(0) -> 1;
factorial(N) when N > 0 -> N * factorial(N - 1).

digits(0) -> 0;
digits(N) when N < 10 -> [N];
digits(N) when N >= 10 -> digits(N div 10) ++ [N rem 10].

solve() ->
  L = digits(factorial(100)),
  lists:sum(L).

File: src/020/p020.exs
defmodule Problem020 do
  defp factorial(0), do: 1
  defp factorial(n) when n > 0, do: n * factorial(n - 1)

  defp sum_factorial_digits(x) do
    factorial(x)
    |> Integer.digits
    |> Enum.sum
  end

  def solve, do: sum_factorial_digits(100)
end

IO.puts Problem020.solve
